src/helpers/url.ts

```diff
@@ -0,0 +1,35 @@
+import { flow } from './fp';
+
+type Query = {
+  [index: string]: string | number | boolean;
+};
+
+export type BuildUrlParams = {
+  pathname: string;
+  query: Query;
+};
+
+const addQueryToUrl = (query: Query) => (url: URL) => {
+  Object.keys(query).forEach(queryKey =>
+    url.searchParams.set(queryKey, query[queryKey].toString()),
+  );
+
+  return url;
+};
+
+const createUrl = (pathname: string) => (apiUrlBase: string) => new URL(pathname, apiUrlBase);
+
+export const buildUrl = ({ pathname, query }: BuildUrlParams) =>
+  flow(createUrl(pathname), addQueryToUrl(query), url => url.toString());
```

Comment:
We don't technically need to return anything in `addQueryToUrl`, since this is just mutating the `URL` that was passed in. It doesn't hurt to return it anyway, but I think if we're using a mutable API, we should make that explicit as possible. `flow` is best suited to immutable functions, so I'm not sure we should use it here.

```suggestion
const addQueryToUrl = (query: Query) => (url: URL) => {
  Object.keys(query).forEach(queryKey =>
    url.searchParams.set(queryKey, query[queryKey].toString()),
  );
};

export const buildUrl = ({ pathname, query }: BuildUrlParams) => (apiBaseUrl: string) => {
  const url = new URL(pathname, apiBaseUrl);
  addQueryToUrl(query)(url);
  return url.toString();
};
```